Scripts/UI/GradientButton.cs

```diff
@@ -92,23 +110,57 @@ public bool visible
 
 				m_Visible = value;
 
+				if (m_Visible && !gameObject.activeSelf)
+					gameObject.SetActive(true);
+
 				this.StopCoroutine(ref m_VisibilityCoroutine);
 				m_VisibilityCoroutine = value ? StartCoroutine(AnimateShow()) : StartCoroutine(AnimateHide());
 			}
 		}
 		bool m_Visible;
 
+		Vector3 originalScale
+		{
+			get { return invertHighlightScale ? m_HighlightContentContainerLocalScale : m_OriginalContentContainerLocalScale; }
+		}
+
+		Vector3 highlightScale
+		{
+			get { return invertHighlightScale ? m_OriginalContentContainerLocalScale : m_HighlightContentContainerLocalScale; }
+		}
+
+		public Sprite icon
+		{
+			set
+			{
+				m_Icon.sprite = value;
+				m_OriginalIconSprite = value;
+			}
+		}
+
+		public bool invertHighlightScale { private get; set; }
```

Comment:
InvertHighlightScale is never set. What is/was it for?

Reply:
There were button state changes that called for this in the early ToolFlow dev phase.